narwhals/_arrow/series.py

```diff
@@ -667,6 +673,13 @@ def clip(
     def to_arrow(self: Self) -> pa.Array:
         return self._native_series.combine_chunks()
 
+    def mode(self: Self) -> ArrowSeries:
+        plx = self.__narwhals_namespace__()
+        col_token = generate_unique_token(n_bytes=8, columns=[self.name])
+        return self.value_counts(name=col_token, normalize=False).filter(
+            plx.col(col_token) == plx.col(col_token).max()
+        )[self.name]
```

Comment:
I was not able to make it work using [`pyarrow.compute.mode`](https://arrow.apache.org/docs/python/generated/pyarrow.compute.mode.html) directly as it requires to know how many distinct most-common values to return.